src/python_bindings/bindings.cpp

```diff
@@ -409,13 +396,14 @@ class PyTiered_HNSWIndex : public PyTieredIndex {
                                 const TieredHNSWParams &tiered_hnsw_params, size_t buffer_limit) {
 
         // Create primaryIndexParams and specific params for hnsw tiered index.
-        VecSimParams primary_index_params = {.algo = VecSimAlgo_HNSWLIB, .hnswParams = hnsw_params};
+        VecSimParams primary_index_params = {.algo = VecSimAlgo_HNSWLIB,
+                                             .algoParams = {.hnswParams = HNSWParams{hnsw_params}}};
 
         // Create TieredIndexParams. Assume that tieredIndexMock::ctx was allocated in the base
         // class constructor, and that the thread pool was created.
         TieredIndexParams tiered_params = {
-            .jobQueue = &tieredIndexMock::jobQ,
-            .jobQueueCtx = tieredIndexMock::ctx,
+            .jobQueue = &(this->mock_thread_pool.jobQ),
+            .jobQueueCtx = this->mock_thread_pool.ctx,
             .submitCb = tieredIndexMock::submit_callback,
             .flatBufferLimit = buffer_limit,
         };
```

Comment:
I suggest moving this to the base class under `getTieredIndexParams() ` or `generateTieredIndexParams() `